addons/survey/controllers/main.py

```diff
@@ -728,39 +723,137 @@ def _get_user_input_domain(self, survey, line_filter_domain, **post):
+
+            answers = request.env['survey.question.answer'].browse(list(filter(bool, ans_ids+row_ids)))  # exclude '0'
+            # For performance, caches all useful columns of survey_question_answer in first query
+            answers_only = answers.filtered(lambda a: not a.matrix_question_id)   # not row_ids
+
+            for answer in answers_only:
+                for row_id in answer_cols_to_rows_mapping[answer.id]:
+                    filters_domains += self._prepare_matching_answer_domain(answer.question_type, answer, row_id=row_id)
+                    search_filters.append(self._prepare_search_filter('A', answer, row_id=row_id))
+
+        # Char_box, Text_box, Numerical_box, Date, Datetime filters
+        if answer_user_input_line_ids:
+            user_input_line_ids = request.env['survey.user_input.line'].browse(answer_user_input_line_ids)
+            for input_line_id in user_input_line_ids:
+                value, operator = self._get_value_with_operator(input_line_id)
+                filters_domains += self._prepare_matching_answer_domain(input_line_id.answer_type, input_line_id, operator=operator, value=value)
+                search_filters.append(self._prepare_search_filter('L', input_line_id, value=value))
+
+        user_input_domain = self._get_results_page_user_input_domain(survey, **post)
+        user_input_lines = self._get_matching_user_input_lines(filters_domains, user_input_domain)
 
         return user_input_lines, search_filters
+
+    def _get_matching_user_input_lines(self, filters_domains, user_input_domain):
+        """
+            Retrieve the matching user input lines depending on the survey base domain and the filters domains
+            :rtype: recordset
+        """
+        # Add filters domain to the base domain
+        if filters_domains:
+            all_required_lines_domains = [
+                [('user_input_line_ids', 'in', request.env['survey.user_input.line'].sudo()._search(filters_domain))]
+                for filters_domain in filters_domains
+            ]
+            user_input_domain = expression.AND([user_input_domain, *all_required_lines_domains])
+
+        # Get the matching user input lines
+        user_inputs_query = request.env['survey.user_input'].sudo()._search(user_input_domain)
+        user_input_lines = request.env['survey.user_input.line'].search([('user_input_id', 'in', user_inputs_query)])
+
+        return user_input_lines
+
+    @staticmethod
+    def _get_filters_from_post(post):
+        """
+            Extract the filters from post depending on the model that needs to be called to retrieve the filtered answer data.
+            Model = 'A' : Answer data in the survey.question.answer model (simple_choice, multiple_choice, matrix)
+            Model = 'L' : Answer data in the survey.user_input.line model (char_box, text_box, numerical_box, date, datetime)
+
+            :rtype: (defaultdict[List[int]], List[int])
+        """
```

Comment:
General and specific comments about the docstrings :)
If we add blank lines, it's better to use them to be also useful when docstring is formatted (with sphinx, markdown).
In practice, newlines before and after lists, before `:params:`/`:return:` block.
The extra indentation does not add anything imo, why make longer lines?

About the return type, the first returned value will be a mapping (here defaultdict) of `int` to `list of int`, so the syntax for the mapping is `[int, List[int]]`. 

If you want to specify it is a defaultdict, you should then use the "full" identifier of the type, so

```suggestion
    def _get_filters_from_post(post):
        """ Extract the filters from post depending on the model that needs to be called to retrieve the filtered answer data.
            
        * Model = 'A' : Match a `survey.question.answer` record (simple_choice, multiple_choice, matrix) 
        * Model = 'L' : Match a `survey.user_input.line` record (char_box, text_box, numerical_box, date, datetime)

        :rtype: (collections.defaultdict[int, list[int]], list[int])
        """
```